pkg/scheduler/core/generic_scheduler.go

```diff
@@ -461,8 +461,9 @@ func (g *genericScheduler) findNodesThatFit(pod *v1.Pod, nodes []*v1.Node) ([]*v
 	if len(g.predicates) == 0 {
 		filtered = nodes
 	} else {
-		allNodes := int32(g.cache.NodeTree().NumNodes())
-		numNodesToFind := g.numFeasibleNodesToFind(allNodes)
+		var lastIndex int
```

Comment:
With lastIndex alone in the class doesn't necessarily give us the continuation since the allNodes may be recomputed between cycles.

We need to remember the reference to the node where the next cycle should start from.